yarl/_url.py

```diff
@@ -32,24 +32,27 @@ class cached_property:
     effectively replacing the function it decorates with an instance
     variable.  It is, in Python parlance, a data descriptor.
 
+    This is trimmed down version of the `cached_property` decorator from
+    the standard library which does not have the undocumented locking
+    behavior that was removed in Python 3.12.
     """
 
+    # When support for Python 3.11 is dropped, this can be replaced with
+    # functools.cached_property.
+
     def __init__(self, wrapped):
         self.wrapped = wrapped
-        try:
-            self.__doc__ = wrapped.__doc__
-        except AttributeError:  # pragma: no cover
-            self.__doc__ = ""
```

Comment:
This was actually unreachable as __doc__ will never not be there https://github.com/python/cpython/blob/d5abd02f36bbee5720944b8906a118a8fb66d75b/Lib/functools.py#L966